### PR TITLE
make use of BKQuery in "running DaVinci on the grid" ?

### PR DESCRIPTION
**Pull request by andriiusachov**
According to my notes from the meeting, there is a suggestion to move to BKQuery in this lesson. 
Isn't it?

### CODE REVIEW

File: first-analysis-steps/davinci-grid.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs.
 
 Before creating your first `ganga` job, open the script `ntuple_options.py`, obtained in the [previous lesson](minimal-dv-job.html), and comment out the lines taking the local input data: we will now use the data stored on grid.
 
-Then, open the file `MC_2016_27163002_Beam6500GeV2016MagDownNu1.625nsPythia8_Sim09b_Trig0x6138160F_Reco16_Turbo03_Stripping28NoPrescalingFlagged_ALLSTREAMS.DST.py`, scroll to the very end, and check if two last lines have information about the `FileCatalog`: if yes, comment them out. They will mislead `ganga` since it only needs to know about the list of LFNs.
+Also, you need to know the path to your data from Bookkeeping. In our case the path is `/MC/2016/Beam6500GeV-2016-MagDown-Nu1.6-25ns-Pythia8/Sim09c/Trig0x6138160F/Reco16/Turbo03/Stripping28r1NoPrescalingFlagged/27163002/ALLSTREAMS.DST`
```
Comment (vlisovsk) [inline first-analysis-steps/davinci-grid.md]: Probably we want to point out here that Event type should be located in the end (which is not the case if you browse the bkk by EventType)

File: first-analysis-steps/davinci-grid.md [hunk 2]
```diff
@@ -37,7 +37,9 @@ j = Job(name='First ganga job')
 myApp = prepareGaudiExec('DaVinci','v44r6', myPath='.')
 j.application = myApp
 j.application.options = ['ntuple_options.py']
-j.application.readInputData('MC_2016_27163002_Beam6500GeV2016MagDownNu1.625nsPythia8_Sim09b_Trig0x6138160F_Reco16_Turbo03_Stripping28NoPrescalingFlagged_ALLSTREAMS.DST.py')
+bkPath = '/MC/2016/Beam6500GeV-2016-MagDown-Nu1.6-25ns-Pythia8/Sim09c/Trig0x6138160F/Reco16/Turbo03/Stripping28r1NoPrescalingFlagged/27163002/ALLSTREAMS.DST'
+data  = BKQuery(bkPath,dqflag=['OK']).getDataset()
```
Comment (alexpearce) [inline first-analysis-steps/davinci-grid.md]: Please add a space after the comma here, `bkPath, dqflag`.

File: first-analysis-steps/davinci-grid.md [hunk 3]
```diff
@@ -37,7 +37,9 @@ j = Job(name='First ganga job')
...
+j.inputdata=data[0:2]     # access only the first 2 files of data
```
Comment (alexpearce) [inline first-analysis-steps/davinci-grid.md]: Spaces around the equals, `inputdata = data`

File: first-analysis-steps/davinci-grid.md [hunk 4]
```diff
@@ -47,7 +49,7 @@ with the option files given in `j.application.options` using a
 backend called `Dirac`, which is "the grid". Instead of specifying the
 files to process as part of the options file you have now to tell the
 `Job` about it. This allows `ganga` to split your job up,
-processing different files simultaneously.
+processing different files simultaneously. Note that data will be accessed using its path in the bookeeping `bkPath`. In order to speed-up, only the first 2 elements (files) of `data` will be accessed.
```
Comment (alexpearce) [inline first-analysis-steps/davinci-grid.md]: * Two 'k's in bookkeeping.
* "In order to speed up _our job_, ... will be accessed; we don't need to look at much data here."

File: first-analysis-steps/davinci-grid.md [hunk 5]
```diff
@@ -99,7 +101,9 @@ myApp = GaudiExec()
 myApp.directory = "./DaVinciDev_v44r6"
 j.application = myApp
 j.application.options = ['ntuple_options.py']
-j.application.readInputData('MC_2016_27163002_Beam6500GeV2016MagDownNu1.625nsPythia8_Sim09b_Trig0x6138160F_Reco16_Turbo03_Stripping28NoPrescalingFlagged_ALLSTREAMS.DST.py')
+bkPath = '/MC/2016/Beam6500GeV-2016-MagDown-Nu1.6-25ns-Pythia8/Sim09c/Trig0x6138160F/Reco16/Turbo03/Stripping28r1NoPrescalingFlagged/27163002/ALLSTREAMS.DST'
+data  = BKQuery(bkPath,dqflag=['OK']).getDataset()
+j.inputdata=data[0:2]
```
Comment (alexpearce) [inline first-analysis-steps/davinci-grid.md]: Same style comments here as [above](https://github.com/lhcb/starterkit-lessons/pull/146/files#diff-fd78835c0bb2d3dca27d71f99e6346a2R41).

File: first-analysis-steps/davinci-grid.md (hunk 1 above)
Comment (andriiusachov) [reply first-analysis-steps/davinci-grid.md]: Done, please have a look

File: first-analysis-steps/davinci-grid.md (hunk 2 above)
Comment (andriiusachov) [reply first-analysis-steps/davinci-grid.md]: Thanks, all done!

File: first-analysis-steps/davinci-grid.md [hunk 6]
```diff
@@ -99,7 +101,9 @@ myApp = GaudiExec()
 myApp.directory = "./DaVinciDev_v44r6"
 j.application = myApp
 j.application.options = ['ntuple_options.py']
-j.application.readInputData('MC_2016_27163002_Beam6500GeV2016MagDownNu1.625nsPythia8_Sim09b_Trig0x6138160F_Reco16_Turbo03_Stripping28NoPrescalingFlagged_ALLSTREAMS.DST.py')
+bkPath = '/MC/2016/Beam6500GeV-2016-MagDown-Nu1.6-25ns-Pythia8/Sim09c/Trig0x6138160F/Reco16/Turbo03/Stripping28r1NoPrescalingFlagged/27163002/ALLSTREAMS.DST'
+data  = BKQuery(bkPath,dqflag=['OK']).getDataset()
```
Comment (chrisburr) [inline first-analysis-steps/davinci-grid.md]: There should be a space here to match with the Python style guide ([PEP8](https://www.python.org/dev/peps/pep-0008/)) and be consistent with the previous use:

```suggestion
data  = BKQuery(bkPath, dqflag=['OK']).getDataset()
```

**Comment by chrisburr** [comment]
Thanks @andriiusachov! 👍

**Pull request merged**